File: src/main/kotlin/at/fhooe/mc/wifipositioning/model/simulation/recording/DataSnapshot.java
package at.fhooe.mc.wifipositioning.model.simulation.recording;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class DataSnapshot {
    private List<WLanData> wLanDataList;
    private long timeStamp;
    private int wayPointID;
    private String routeIdentifier;

    public DataSnapshot(List<WLanData> wLanDataList, long timeStamp) {
        this.wLanDataList = wLanDataList;
        this.timeStamp = timeStamp;
    }

    public DataSnapshot(String routeIdentifier, int wayPointID, List<WLanData> wLanDataList, long timeStamp) {
        this.wLanDataList = wLanDataList;
        this.timeStamp = timeStamp;
        this.wayPointID = wayPointID;
        this.routeIdentifier = routeIdentifier;
    }

    public DataSnapshot(){}


    public List<WLanData> getwLanDataList() {
        return wLanDataList;
    }

    @JsonProperty("WLanData")
    public void setwLanDataList(List<WLanData> wLanDataList) {
        this.wLanDataList = wLanDataList;
    }

    public long getTimeStamp() {
        return timeStamp;
    }

    @JsonProperty("TimeStamp")
    public void setTimeStamp(long timeStamp) {
        this.timeStamp = timeStamp;
    }

    public int getWayPointID() {
        return wayPointID;
    }

    @JsonProperty("WaypointID")
    public void setWayPointID(int wayPointID) {
        this.wayPointID = wayPointID;
    }

    public String getRouteIdentifier() {
        return routeIdentifier;
    }

    @JsonProperty("RouteIdentifier")
    public void setRouteIdentifier(String routeIdentifier) {
        this.routeIdentifier = routeIdentifier;
    }
}

File: src/main/kotlin/at/fhooe/mc/wifipositioning/model/recording/DataSnapshot.kt
package at.fhooe.mc.wifipositioning.model.recording

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * A snapshot during a recording containing multiple Wi-Fi data of multiple detected SSIDs
 *
 * @property wifiDataList a list of captured Wi-Fi data
 * @property timestamp the timestamp of the recording
 * @property wayPointID the associated waypoint ID of the recording
 * @property routeIdentifier the ID of the route
 */
data class DataSnapshot(@JsonProperty("WLanData") val wifiDataList: List<WifiData>,
                        @JsonProperty("TimeStamp") val timestamp: Long,
                        @JsonProperty("WaypointID") val wayPointID: Int,
                        @JsonProperty("RouteIdentifier") val routeIdentifier: String)